plutus-contract/test/Spec/TxConstraints/MustMint.hs

```diff
@@ -0,0 +1,275 @@
+import Ledger.Constraints.OnChain.V1 qualified as Constraints (checkScriptContext)
+import Ledger.Constraints.TxConstraints qualified as Constraints (collectFromTheScript, mustMintCurrency,
+                                                                  mustMintCurrencyWithRedeemer, mustMintValue,
+                                                                  mustMintValueWithRedeemer, mustPayToTheScript)
+import Ledger.Test (coinMintingPolicy)
+import Ledger.Tx qualified as Tx
+import Ledger.Typed.Scripts qualified as Scripts
+import Ledger.Value (CurrencySymbol (CurrencySymbol), TokenName (TokenName))
+import Plutus.Contract as Con
+import Plutus.Contract.Test (assertContractError, assertFailedTransaction, assertValidatedTransactionCount,
+                             checkPredicateOptions, defaultCheckOptions, w1, (.&&.))
+import Plutus.Script.Utils.V1.Scripts qualified as PSU.V1
+import Plutus.Trace qualified as Trace
+import Plutus.V1.Ledger.Api (MintingPolicyHash, Redeemer, ToData (toBuiltinData),
+                             UnsafeFromData (unsafeFromBuiltinData))
+import Plutus.V1.Ledger.Scripts (ScriptError (EvaluationError), unitRedeemer)
+import Plutus.V1.Ledger.Value qualified as Value
+import PlutusTx qualified
+import Prelude hiding (not)
+
+tests :: TestTree
+tests =
+    testGroup "MustMint"
+        [ mustMintCurrencyWithRedeemerSuccessfulMint
+        , mustMintCurrencyWithRedeemerMissingPolicyLookup
+        , mustMintCurrencyWithRedeemerPhase2Failure
+        , mustMintCurrencySuccessfulMint
+        , mustMintValueWithRedeemerSuccessfulMint
+        , mustMintValueSuccessfulMint
+        ]
+
+trace ::  Contract () Empty ContractError () -> Trace.EmulatorTrace ()
+trace contract = do
+    void $ Trace.activateContractWallet w1 contract
+    void $ Trace.waitNSlots 1
+
+data UnitTest
+instance Scripts.ValidatorTypes UnitTest
+
+tknName :: TokenName
+tknName = "TokenB"
+
+tknAmount :: Integer
+tknAmount = 21_000_000
+
+tknValue :: Ledger.Value
+tknValue = Value.singleton coinMintingPolicyCurrencySymbol tknName tknAmount
+
+coinMintingPolicyHash :: MintingPolicyHash
+coinMintingPolicyHash = PSU.V1.mintingPolicyHash coinMintingPolicy
+
+coinMintingPolicyCurrencySymbol :: CurrencySymbol
+coinMintingPolicyCurrencySymbol = CurrencySymbol $ unsafeFromBuiltinData $ toBuiltinData coinMintingPolicyHash
+
+-- Uses onchain and offchain constraint mustMintCurrencyWithRedeemer to mint tokens
+mustMintCurrencyWithRedeemerSuccessfulMint :: TestTree
+mustMintCurrencyWithRedeemerSuccessfulMint =
+    checkPredicateOptions
+    defaultCheckOptions
+    "Successful spend of tokens using mustMintCurrencyWithRedeemer"
+    (assertValidatedTransactionCount 2)
+    (void $ trace $ mustMintCurrencyWithRedeemerContract tknName)
+
+-- Uses onchain and offchain constraint mustMintCurrencyWithRedeemer but with a contract that is missing lookup for the minting policy, asserts contract error.
```

Comment:
Do `-- | Uses on chain...` as comment to be compatible with Haddock. Not a huge problem since these are test modules, but just for consistency.

Reply:
Have applied this change but doesn't affect the built Haddock that I can see. Perhaps the test modules are excluded.